src/reformat/alpha.rs

```diff
@@ -82,8 +90,12 @@ impl rgb::Image {
         clamp_u16(alpha, 0, dst_max_channel)
     }
 
-    pub fn reformat_alpha(&mut self, image: &image::Image) -> AvifResult<()> {
-        if !self.has_alpha() {
+    pub fn import_alpha_from(&mut self, image: &image::Image) -> AvifResult<()> {
+        if !self.has_alpha()
+            || !image.has_alpha()
+            || self.width != image.width(Plane::A) as u32
```

Comment:
you can simply use `image.width` here since alpha plane is never subsampled.

ditto for height.